src/zenml/client.py

```diff
@@ -3140,6 +3147,331 @@ def list_run_metadata(
+            updated: Use the last updated date for filtering
+            user_id: The  id of the user to filter by.
+            name: The name of the secret to filter by.
+            scope: The scope of the secret to filter by.
+
+        Returns:
+            A list of all the secret models.
+        """
+        secret_filter_model = SecretFilterModel(
+            page=page,
+            size=size,
+            sort_by=sort_by,
+            logical_operator=logical_operator,
+            user_id=user_id,
+            name=name,
+            scope=scope,
+            id=id,
+            created=created,
+            updated=updated,
+        )
+        secret_filter_model.set_scope_workspace(self.active_workspace.id)
+        return self.zen_store.list_secrets(
+            secret_filter_model=secret_filter_model
+        )
+
+    def list_secrets_in_scope(
+        self,
+        scope: SecretScope,
+    ) -> Page[SecretResponseModel]:
+        """Fetches the list of secret in a given scope.
+
+        Args:
+            scope: The secrets scope to search for.
+
+        Returns:
+            The list of secrets.
+        """
+        logger.debug(f"Fetching the secrets in scope {scope.value}.")
+
+        return self.list_secrets(
+            scope=scope,
+        )
+
+    def update_secret(
+        self,
+        name_id_or_prefix: str,
+        scope: Optional[SecretScope] = None,
+        new_name: Optional[str] = None,
+        new_scope: Optional[SecretScope] = None,
+        add_or_update_values: Optional[Dict[str, str]] = None,
+        remove_values: Optional[List[str]] = None,
+    ) -> SecretResponseModel:
+        """Updates a secret.
+
+        Args:
+            name_id_or_prefix: The name, id or prefix of the id for the
+                secret to update.
+            scope: The scope of the secret to update.
+            new_name: The new name of the secret.
+            new_scope: The new scope of the secret.
+            add_or_update_values: The values to add or update.
+            remove_values: The values to remove.
+
+        Returns:
+            The updated secret.
+        """
+        secret = self.get_secret(
+            name_id_or_prefix=name_id_or_prefix,
+            scope=scope,
+            # Don't allow partial name matches, but allow partial ID matches
+            allow_partial_name_match=False,
+            allow_partial_id_match=True,
+        )
+
+        secret_update = SecretUpdateModel()
+
+        if new_name:
+            secret_update.name = new_name
+        if new_scope:
+            secret_update.scope = new_scope
+        values: Dict[str, Optional[str]] = {}
+        if add_or_update_values:
+            values.update(add_or_update_values)
+        if remove_values:
+            for key in remove_values:
+                values[key] = None
+        if values:
+            secret_update.values = cast(Dict[str, Optional[SecretStr]], values)
```

Comment:
This cast is incorrect right, it's not actually a dict containing secret strings? Do we run into some trouble of calling `SecretStr.some_method()` somewhere down the line by doing this?

Reply:
Quite right, yes. You found another bug that likely would've been revealed by the integration tests I need to write :)